### PR TITLE
Add TTL and Container modes to BaseDataAccessor and its implementations

### LINKED ISSUE #2081: API methods in zookeeper-api don't support extended types
**Is your feature request related to a problem? Please describe.**
Zookeeper versions 3.5.3+ support some extended types for creating ZNodes, namely CONTAINER, PERSISTENT_WITH_TTL, and PERSISTENT_SEQUENTIAL_WITH_TTL. However, API methods that can support these types haven't been added to zookeeper-api module.

**Describe the solution you'd like**
Add API methods for Zookeeper extended types to zookeeper-api.

### PR DESCRIPTION
**Pull request by rabashizade**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #2081 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR adds support for TTL and Container modes to BaseDataAccessor and its implementations by taking advantage of relevant API from ZkClient and its descendent classes.

### Tests

- [x] The following tests are written for this issue:

Two tests (`testSyncCreateWithTTL()` and `testSyncCreateContainer()`) were added to `TestZkBaseDataAccessor`. In addition, `testAsyncZkBaseDataAccessor()` in the same class was modified to test the newly added API.

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/BaseDataAccessor.java [hunk 1]
```diff
@@ -43,6 +43,18 @@
...
+   * This will always attempt to create the znode, if it exists it will return false. Will
+   * create parents if they do not exist. For performance reasons, it may try to create
+   * child first and only if it fails it will try to create parents
+   * @param path path to the ZNode to create
+   * @param record the data to write to the ZNode
+   * @param options Set the type of ZNode see the valid values in {@link AccessOption}
+   * @param ttl TTL of the node in milliseconds, if options supports it
+   * @return true if creation succeeded, false otherwise (e.g. if the ZNode exists)
+   */
+  boolean create(String path, T record, int options, long ttl);
```
Comment (junkaixue) [inline helix-core/src/main/java/org/apache/helix/BaseDataAccessor.java:56]: So this only applies to create? It means this property cannot be added in the middle by override or set?

**Review by junkaixue — COMMENTED** [review]
Overall, I dont any other concerns.

**Review by qqu0127 — APPROVED** [review]
LGTM, please address other comments. thanks

**Comment by rabashizade** [comment]
This PR is ready to be merged, approved by @qqu0127.

**Pull request merged**